nixos/modules/services/misc/gitea.nix

```diff
@@ -0,0 +1,405 @@
+      description = "Go Git Service";
+      home = cfg.stateDir;
+      createHome = true;
+      shell = pkgs.bash;
+    };
+
+    users.extraGroups = optional (cfg.group == giteaGroup) {
+      name = giteaGroup;
+      gid = config.ids.gids.gitea;
+    };
+
+    services.postgresql = mkIf useLocalPostgresqlDatabase {
+      enable = true;
+    };
+
+    system.activationScripts.gitea = ''
+      mkdir -p ${cfg.stateDir}
+      chown -R ${cfg.user}:${cfg.group} ${cfg.stateDir}
+    '';
+
+    systemd.services.gitea = {
+      description = "Gitea (Go Git Service)";
+
+      wantedBy = [ "multi-user.target" ];
+      requires = [ "network.target" ]
+        ++ optional useLocalPostgresqlDatabase "postgresql.service";
+      after = [ "network.target" ]
+        ++ optional useLocalPostgresqlDatabase "postgresql.service";
+
+      path = [ pkgs.gitea.bin ];
+
+      environment = {
+        USER = cfg.user;
+        HOME = cfg.stateDir;
+        GITEA_WORK_DIR = cfg.stateDir;
+      };
+
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+        WorkingDirectory = cfg.stateDir;
+        ExecStart = "${pkgs.gitea.bin}/bin/gitea web";
+        Restart = "always";
+        PermissionsStartOnly = true;
+        PrivateDevices = true;
+        PrivateTmp = mkIf (!useLocalPostgresqlDatabase) true;
+        JoinsNamespaceOf = mkIf useLocalPostgresqlDatabase "postgresql.service";
+      };
+
+      preStart = ''
+        # copy custom configuration and generate a random secret key if needed
+        ${optionalString (cfg.useWizard == false) ''
+          mkdir -p ${cfg.stateDir}/custom/conf
+          cp -f ${configFile} ${cfg.stateDir}/custom/conf/app.ini
+          KEY=$(head -c 16 /dev/urandom | tr -dc A-Za-z0-9)
+          sed -i "s,#secretkey#,$KEY,g" ${cfg.stateDir}/custom/conf/app.ini
+        ''}
+
+        mkdir -p ${cfg.repository.root}
+        # update all hooks' binary paths
+        HOOKS=$(find ${cfg.repository.root} -mindepth 4 -maxdepth 4 -type f -wholename "*git/hooks/*")
+        if [ "$HOOKS" ]
+        then
+          sed -ri 's,/nix/store/[a-z0-9.-]+/bin/gitea,${pkgs.gitea.bin}/bin/gitea,g' $HOOKS
+          sed -ri 's,/nix/store/[a-z0-9.-]+/bin/env,${pkgs.coreutils}/bin/env,g' $HOOKS
+          sed -ri 's,/nix/store/[a-z0-9.-]+/bin/bash,${pkgs.bash}/bin/bash,g' $HOOKS
+          sed -ri 's,/nix/store/[a-z0-9.-]+/bin/perl,${pkgs.perl}/bin/perl,g' $HOOKS
+        fi
+
+        # setup postgres user
+        ${optionalString useLocalPostgresqlDatabase ''
+          if ! [ -e ${cfg.stateDir}/.db-created ]; then
+            ${pkgs.postgresql}/bin/psql \
+              postgres -c "CREATE USER ${cfg.database.user} WITH PASSWORD '${cfg.database.password}';";
+            ${pkgs.postgresql}/bin/createdb \
+              ${cfg.database.name} -O ${cfg.database.user}
+            touch ${cfg.stateDir}/.db-created
+          fi
+        ''}
+      chown -R ${cfg.user}:${cfg.group} ${cfg.stateDir}
```

Comment:
Here you `chown` again. I would remove `activationsScripts` and create `stateDir` at the beginning of `preStart`.